homeassistant/components/avri/sensor.py

```diff
@@ -0,0 +1,106 @@
+"""Support for Avri waste curbside collection pickup."""
+from datetime import timedelta
+import logging
+
+from avri.api import Avri, AvriException
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+CONF_COUNTRY_CODE = "country_code"
+CONF_POSTCODE = "postcode"
+CONF_HOUSE_NUMBER = "house_number"
+CONF_HOUSE_NUMBER_EXTENSION = "house_number_extension"
+DEFAULT_NAME = "avri"
+ICON = "mdi:trash-can-outline"
+SCAN_INTERVAL = timedelta(hours=4)
+DEFAULT_COUNTRY_CODE = "NL"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_POSTCODE): cv.string,
+        vol.Required(CONF_HOUSE_NUMBER): cv.positive_int,
+        vol.Optional(CONF_HOUSE_NUMBER_EXTENSION): cv.string,
+        vol.Optional(CONF_COUNTRY_CODE, default=DEFAULT_COUNTRY_CODE): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Avri Waste platform."""
+    client = Avri(
+        postal_code=config[CONF_POSTCODE],
+        house_nr=config[CONF_HOUSE_NUMBER],
+        house_nr_extension=config[CONF_HOUSE_NUMBER_EXTENSION],
+        country_code=config[CONF_COUNTRY_CODE],
+    )
+
+    try:
+        each_upcoming = client.upcoming_of_each()
+    except AvriException as ex:
+        raise PlatformNotReady from ex
+    else:
+        entities = [
+            AvriWasteUpcoming(config[CONF_NAME], client, upcoming.name)
+            for upcoming in each_upcoming
+        ]
+        add_entities(entities, True)
+
+
+class AvriWasteUpcoming(Entity):
+    """Avri Waste Sensor."""
+
+    def __init__(self, name: str, client: Avri, waste_type: str):
```

Comment:
```suggestion
    def __init__(self, name, client, waste_type):
```

Reply:
I absolutely love this python3.5+ typehints introduced in https://www.python.org/dev/peps/pep-0484/. It helps the readers of the code, and also any modern editor with developing.

Is there a particular reason they should be removed?

Reply:
Well I just wanted it to be consistent with other methods in that class but I think it is fine if they stay in this case.